src/oscar/apps/basket/formsets.py

```diff
@@ -23,6 +23,10 @@ def _should_delete_form(self, form):
         """
         if super()._should_delete_form(form):
             return True
+
+        # If related form instance already removed, let's skip this form.
```

Comment:
This comment is a little misleading - we're not skipping the form, but deleting it.

Reply:
Fixed.